src/ILCompiler.Compiler/src/Compiler/Compilation.cs

```diff
@@ -214,7 +214,7 @@ public MethodDesc ResolveVirtualMethod(MethodDesc declMethod, TypeDesc implType)
             return _devirtualizationManager.ResolveVirtualMethod(declMethod, implType);
         }
 
-        public bool NeedsRuntimeLookup(ReadyToRunHelperId lookupKind, object targetOfLookup)
+        public virtual bool NeedsRuntimeLookup(ReadyToRunHelperId lookupKind, object targetOfLookup)
```

Comment:
Maybe it would be okay for the R2R part not to call this API at all. It already forked its own version of the ComputeConstantLookup and ComputeGenericLookup below. I don't think it makes sense to try to keep it call this one on Compilation when it cannot call the ones below either.